### PR TITLE
Register navigation sample implementation

### PR DESCRIPTION
**Pull request by almirvuk**
I have implemented this RegisterForNavigation sample with different pages and viewmodels.

Page(s):

- AnotherPage
- BasicPage
- DroidPage
- FullRegisteredPage
- IOSPage
- OtherIdiomPage
- OtherPlatformPage
- PhonePage
- TabletPage
- UWPPage

ViewModel(s):

- BaseViewModel
- AnotherPageViewModel
- BasicPageViewModel
- FullRegistratedPageViewModel
- IdiomSharedViewModel
- PlatformSharedViewModel

### CODE REVIEW

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/BaseViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Prism.Mvvm;
+
+namespace PrismSample.ViewModels
+{
+    public class BaseViewModel : BindableBase
+    {
+        private string pageText;
+        public string PageText
```
Comment (dansiegel) [inline 21-RegisterForNavigation/src/PrismSample/ViewModels/BaseViewModel.cs]: let's rename this to just Text....

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/FullRegistratedPageViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace PrismSample.ViewModels
+{
+    public class FullRegistratedPageViewModel : BaseViewModel
+    {
+        public FullRegistratedPageViewModel()
+        {
+            PageText = "MVVM and data binding is working with full registered page and View-Model!";
```
Comment (dansiegel) [inline 21-RegisterForNavigation/src/PrismSample/ViewModels/FullRegistratedPageViewModel.cs]: ViewModel => !View-Model

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/IdiomSharedViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace PrismSample.ViewModels
+{
+    public class IdiomSharedViewModel : BaseViewModel
+    {
+        public IdiomSharedViewModel()
+        {
+            PageText = "MVVM and data binding is working with IDIOM shared View-Model!";
```
Comment (dansiegel) [inline 21-RegisterForNavigation/src/PrismSample/ViewModels/IdiomSharedViewModel.cs]: ViewModel => !View-Model
Idiom => !IDIOM

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/MainMenuPageViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        public MainMenuPageViewModel(INavigationService navigationService)
+        {
+            _navigationService = navigationService;
+
+            GoToBasicPageCommand = new DelegateCommand(GoToBasicPage);
+            GoToIdiomSpecificPageCommand = new DelegateCommand(GoToIdiomSpecificPage);
+            GoToFullRegistratedPageCommand = new DelegateCommand(GoToFullRegistratedPage);
+            GoToPlatformSpecificPageCommand = new DelegateCommand(GoToPlatformSpecificPage);
+            GoToCustomNameRegisteredPageCommand = new DelegateCommand(GoToCustomNameRegisteredPage);
+        }
```
Comment (dansiegel) [inline 21-RegisterForNavigation/src/PrismSample/ViewModels/MainMenuPageViewModel.cs]: Replace this madness with a single NavigateCommand:

```cs
public DelegateCommand<string> NavigateCommand { get; }

public async void OnNavigateCommandExecuted(string uri) =>
    await _navigationService.NavigateAsync(uri);
```

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/PlatformSharedViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace PrismSample.ViewModels
+{
+    public class PlatformSharedViewModel : BaseViewModel
+    {
+        public PlatformSharedViewModel()
+        {
+            PageText = "MVVM and data binding is working with platform shared View-Model!";
```
Comment (dansiegel) [inline 21-RegisterForNavigation/src/PrismSample/ViewModels/PlatformSharedViewModel.cs]: ViewModel => !View-Model

File: 21-RegisterForNavigation/src/PrismSample/ViewModels/MainMenuPageViewModel.cs (hunk 4 above)
Comment (almirvuk) [reply 21-RegisterForNavigation/src/PrismSample/ViewModels/MainMenuPageViewModel.cs]: You are 100% right about this! In my defence, I was writing this in a train going back from hometown 😂

**Comment by almirvuk** [comment]
Made some changes based on the last review 😄

**Pull request merged**